DigitalLearningSolutions.Web/Attributes/DelegateOnlyInaccessibleAttribute.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace DigitalLearningSolutions.Web.Attributes
+{
+    using System;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public class DelegateOnlyInaccessibleAttribute : Attribute, IAuthorizationFilter
+    {
+        public void OnAuthorization(AuthorizationFilterContext context)
+        {
+            var user = context.HttpContext.User;
+
+            if (user.Identity.IsAuthenticated && !user.GetAdminId().HasValue)

```

Comment:
This is not particularly important because I think this combination of IsAuthenticated and no AdminId is sufficient, but I'm wondering if we could check the claim LearnUserAuthenticated. This is a bool that is true if the user is logged into a delegate account. Because it is not mutually exclusive with the AdminId, we'd still need to check both. So functionally it wouldn't change but it feels better to me to check LearnUserAuthenticated and AdminId. But it shouldn't make a difference functionally, so I'm happy with either.

Reply:
I think it's better to check it so that it's more robust to errors in the login/account system, otherwise we may risk e.g. an infinite redirect loop.

Reply:
I'm going to fix this in the review for the navigation bar changes, as it's low-priority and to do otherwise would require some annoying git surgery.

Reply:
I think checking for the learnUserAuthenticated sounds good - just to make sure they don't get redirected back from the Learning Portal :)